src/arrayUtils.js

```diff
@@ -0,0 +1,68 @@
+if (typeof maybeExport === 'undefined') {
+	function maybeExport () {}
+}
+
+/** @type {function(string, boolean=, number=)} */
+function intArrayFromString(stringy, dontAddNull, length) {
+  var len = length > 0 ? length : lengthBytesUTF8(stringy)+1;
+  var u8array = new Array(len);
+  var numBytesWritten = stringToUTF8Array(stringy, u8array, 0, u8array.length);
+  if (dontAddNull) u8array.length = numBytesWritten;
+  return u8array;
+}
+{{{ maybeExport('intArrayFromString') }}}
+
+function intArrayToString(array) {
+  var ret = [];
+  for (var i = 0; i < array.length; i++) {
+    var chr = array[i];
+    if (chr > 0xFF) {
+#if ASSERTIONS
+      assert(false, 'Character code ' + chr + ' (' + String.fromCharCode(chr) + ')  at offset ' + i + ' not in 0x00-0xFF.');
```

Comment:
perhaps we can remove this assertion? it would simplify this PR, and I'm not sure how helpful it actually is. The user would likely see a weird string anyhow as a result of this. @juj , what do you think?